cmd/common/image/verify.go

```diff
@@ -0,0 +1,104 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/distribution/reference"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2/registry/remote"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/signature"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/signature/cosign"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/signature/notation"
+)
+
+func NewVerifyCmd() *cobra.Command {
+	var publicKeys []string
+	var certificates []string
+	var policy string
+
+	cmd := &cobra.Command{
+		Use:   "verify [images...]",
+		Short: "Verify the signature of one or more OCI images",
+		Long:  `Verify an image's signature without running it. Supports both Cosign (public keys) and Notation (certificates + trust policy).`,
+
+		Args: cobra.MinimumNArgs(1),
```

Comment:
Let's keep thing simple and allow to verify only image at a time, so:

```suggestion
		Args: cobra.ExactArgs(1),
```